awx/main/utils/receptor.py

```diff
@@ -0,0 +1,77 @@
+import logging
+import yaml
+import time
+
+from receptorctl.socket_interface import ReceptorControl
+
+
+logger = logging.getLogger('awx.main.utils.receptor')
+
+
+def get_receptor_ctl():
+    return ReceptorControl('/var/run/receptor/receptor.sock')
+
+
+def worker_info(node_name):
```

Comment:
This method is focused on the `--worker-info`, but there's a lot of great error handling here that it seems like we should use whenever we invoke `ansible-runner` w/ receptor. Could we turn this into a more generic helper for that?

Reply:
I do hope this inspires some additional error handling for jobs, but it's hard to see code sharing. Here, we start a fire-and-forget job and then poll until it's finished, and then read the entire standard out. That's way more simple than what jobs do